esctest/tests/xterm_winops.py

```diff
@@ -12,288 +16,599 @@
-                 abs(actual_size.height() - desired_size.height()))
-    max_error = 20
-    AssertTrue(error.width() <= max_error)
-    AssertTrue(error.height() <= max_error)
-
-    esccmd.XTERM_WINOPS(esccmd.WINOP_RESIZE_PIXELS,
-                            original_size.height(),
-                            original_size.width())
+    if escargs.args.expected_terminal == "xterm":
+      maximum_size = GetScreenSizePixels()
+      original_size = GetWindowSizePixels()
+      desired_size = Size(self.average_width(maximum_size, original_size),
+                          self.average_height(maximum_size, original_size))
+
+      esccmd.XTERM_WINOPS(esccmd.WINOP_RESIZE_PIXELS,
+                          desired_size.height(),
+                          desired_size.width())
+      self.delay_after_resize()
+      self.check_actual_size_pixels(desired_size)
+
+      esccmd.XTERM_WINOPS(esccmd.WINOP_RESIZE_PIXELS,
+                          original_size.height(),
+                          original_size.width())
+      self.delay_after_resize()
+    else:
+      original_size = GetWindowSizePixels()
+      desired_size = Size(400, 200)
+
+      esccmd.XTERM_WINOPS(esccmd.WINOP_RESIZE_PIXELS,
+                          desired_size.height(),
+                          desired_size.width())
+      # See if we're within 20px of the desired size on each dimension. It won't
+      # be exact because most terminals snap to grid.
+      actual_size = GetWindowSizePixels()
+      error = Size(abs(actual_size.width() - desired_size.width()),
+                   abs(actual_size.height() - desired_size.height()))
+      max_error = 20
+      AssertTrue(error.width() <= max_error)
+      AssertTrue(error.height() <= max_error)
+
+      esccmd.XTERM_WINOPS(esccmd.WINOP_RESIZE_PIXELS,
+                          original_size.height(),
+                          original_size.width())
 
   def test_XtermWinops_ResizePixels_OmittedHeight(self):
     """Resize the window to a pixel size, omitting one parameter. The size
     should not change in the direction of the omitted parameter."""
-    original_size = GetWindowSizePixels()
-    desired_size = Size(400, original_size.height())
-
-    esccmd.XTERM_WINOPS(esccmd.WINOP_RESIZE_PIXELS,
-                            None,
-                            desired_size.width())
-    # See if we're within 20px of the desired size. It won't be exact because
-    # most terminals snap to grid.
-    actual_size = GetWindowSizePixels()
-    error = Size(abs(actual_size.width() - desired_size.width()),
-                 abs(actual_size.height() - desired_size.height()))
-    max_error = 20
-    AssertTrue(error.width() <= max_error)
-    AssertTrue(error.height() <= max_error)
-
-    esccmd.XTERM_WINOPS(esccmd.WINOP_RESIZE_PIXELS,
-                            original_size.height(),
-                            original_size.width())
+    if escargs.args.expected_terminal == "xterm":
```

Comment:
Did you go with separate tests for xterm because iTerm2 doesn't support GetScreenSizePixels()? It would be nice to share the same tests if possible.

Reply:
yes - some merging would be possible, but I didn't do that, to make the differences easier to follow.

Reply:
Once GetScreenSizePixels has been modified to return a constant, it should be possible to reduce the amount separate xterm code in these tests. It would be good to minimize complexity and test the same thing across different terminals.

Reply:
It does this if the terminal isn't xterm:

   gScreenSizePixels = Size(1024, 768)